spec/unit/matchers/run_spec.rb

```diff
@@ -0,0 +1,96 @@
+require 'spec_helper'
+
+describe RSpec::Puppet::FunctionMatchers::Run do
+  it 'should call the lambda with no params' do
+    received_params = nil
+    if Puppet.version.to_f >= 4.0
+      subject.matches?(lambda { |env, *params| received_params = params })
+      expect(received_params).to eq([])
+    else
+      subject.matches?(lambda { |params| received_params = params })
+      expect(received_params).to be_nil
+    end
+  end
+
+  it 'should not match a lambda that raises an error' do
+    expect(subject.matches?(lambda { |env, *params| raise StandardError, 'Forced Error' })).to be_false
+  end
+
+  [ [], [true], [false], [''], ['string'], [nil], [0], [[]], [{}], [1.1], ['one', 'two'] ].each do |supplied_params|
```

Comment:
good call. fixed. Also reordered the values, added a hash with a key, and added `:undef` to all other lists